File: panagram.ts
// function checkPanagram(str) {
//   // remove space
//   const removedSpaceStr = str.replace(/[ .]+/g, "").toLowerCase();

//   // use set to get unique alphabet characters
//   const uniqueSet = new Set(removedSpaceStr.split(""));
//   const uniqueArr = [...uniqueSet];

//   // count the arrag if not equal to 26 return false
//   return uniqueArr.length == 26 ? "YES" : "NO";
// }
function checkPanagram(str: string) {
  const alphabet = "abcdefghijklmnopqrstuvwxyz";
  for (let char of alphabet) {
    if (!str.includes(char)) {
      return false;
    }
  }
  return true;
}


console.log(checkPanagram("The quick brown fox jumps over the lazy dog."));
console.log(checkPanagram("Hello world."));

File: panagram.py
def checkPanagram(str):
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    for char in alphabet:
        if char not in str.lower():
            return False
    return True


print(checkPanagram("The quick brown fox jumps over the lazy dog."))
print(checkPanagram("The quick brown fox jumps over the lazy g."))


